Snabble/UI/AssetProvider.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  AssetProvider.swift
+//  OnboardingUIKit
+//
+//  Created by Uwe Tilemann on 08.08.22.
+//
+import Foundation
+
+#if os(macOS)
+public typealias OSImage = NSImage
+public typealias OSColor = NSColor
+#elseif os(iOS) || os(tvOS) || os(watchOS)
+public typealias OSImage = UIImage
+public typealias OSColor = UIColor
+#endif
+
+public protocol ImageProviding: AnyObject {
+    func image(for name: String, domain: Any?) -> OSImage?
+}
+public protocol ColorProviding: AnyObject {
+    func color(for name: String, domain: Any?) -> OSColor?
+}
+public protocol StringProviding: AnyObject {
+    func string(for name: String, domain: Any?) -> String?
+}
+public protocol UrlProviding: AnyObject {
+    func url(forResource name: String?, withExtension ext: String?, domain: Any?) -> URL?
+}
+
+public typealias AssetProviding = ImageProviding & ColorProviding & StringProviding & UrlProviding
+
+/**
+ `AssetProvider` implementing `AssetProviding`.
+*/
+public final class AssetProvider: AssetProviding {
+    /// A shared instance
+    public static var shared = AssetProvider()
+
+    /// A provider to get resources from
+    public weak var provider: AssetProviding?
+
+    public func color(for name: String, domain: Any? = nil) -> OSColor? {
+        return provider?.color(for: name, domain: domain) ?? OSColor.black
```

Comment:
In allen Methoden fehlt der Fallback. Warum hast du ihn rausgelassen, es aber bereits zum Review gestellt? Sollten wir das AssetProviding Pattern eventuell als separaten Request aufsetzen, damit wir es unabhängig vom Onboarding als Feature abschließen können?

Reply:
Der Fallback ist z.Zt. `?? OSColor.black`, was natürlich nicht ausreichend ist. Da braucht es noch etwas mehr.